File: cpp_pkg/src/resistance_client.cpp
#include "rclcpp/rclcpp.hpp"
#include "robot_interface/srv/resistance_system.hpp"

class ResistanceClient: public rclcpp::Node{

    private:

        std::thread thread_;
        
    public:

        ResistanceClient(float tension, float current): Node("resistance_client"){
            thread_ = std::thread(std::bind(&ResistanceClient::call_server, this, tension, current));
        }

        void call_server(float tension, float current){

            auto client = this->create_client<robot_interface::srv::ResistanceSystem>("resistance_measure");
            auto request = std::make_shared<robot_interface::srv::ResistanceSystem::Request>();

            while (!client->wait_for_service(std::chrono::seconds(1)))
                RCLCPP_WARN(this->get_logger(), "Waiting for service");

            request->tension = tension;
            request->current = current;

            auto future = client->async_send_request(request);

            try{

                auto response = future.get();
                RCLCPP_INFO(this->get_logger(), "Resistance: %.2f ohms", (float)response->resistance);

            }

            catch(std::exception &ex){

                RCLCPP_ERROR(this->get_logger(), "Exception occurred");

            }
        }

};

int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<ResistanceClient>(3.3, 0.125);
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;

}

File: python_packages/python_pkg/python_pkg/resistance_client.py
#!/usr/bin/env python3 

import rclpy
from rclpy.node import Node
from robot_interface.srv import ResistanceSystem
from functools import partial

class ResistanceClient(Node):
    def __init__(self):
        super().__init__("resistance_client")
        
        
    def call_server(self, tension, current):
        self.client_ = self.create_client(ResistanceSystem, "resistance_measure")
        request = ResistanceSystem.Request()
        request.tension = tension
        request.current = current

        while not self.client_.wait_for_service(1):
            self.get_logger().warn("Waiting for service")



        future = self.client_.call_async(request)
        future.add_done_callback(partial(self.callback_call))



    def callback_call(self, future):
        try:
            response = future.result()
            self.get_logger().info(f"Resistance: {response.resistance}")

        except Exception as ex:
            self.get_logger().error("ERROR: Except!!!%r" %(ex,))


    

def main(args = None):
    rclpy.init(args=args)
    node = ResistanceClient()
    node.call_server(3.3, 0.125)
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()